module3/lessons/week3/CreatingRelatedResources.md

```diff
@@ -0,0 +1,349 @@
+* Use a form to create new reviews for a movie
+
+## Nested Resources
+
+So far, we have been working with a single resource - Movie.  This makes our routes relatively straight forward. We learned about RESTful routes and implemented them into our project.
+
+<section class='call-to-action'>
+<strong>To Review</strong>: List all the RESTful routes for Movie in your notebook.
+</section>
+
+RESTful routes can also be used for more complex data relationships.  
+
+We are going to be updating our application so that we have a database similar to the diagram below:
+![](/assets/images/module3/Week3/MovieReviewERD.png)
+
+<section class='answer' markdown="1">
+
+### Restful Routes for a Nested Resource
+If we imagine we have an application that tracks artists and their paintings.  An artist has many paintings, and a painting belongs to an artist.  The 14 restful routes we would create (7 for each resource) could look like:
+
+| path                                  | method     | Controller#Action |  
+| ------------------------------------- | ---------: | ----------------: |  
+| /artists                              | GET        | Artists#Index     |  
+| /artists/:id                          | GET        | Artists#Show      |  
+| /artists/new                          | GET        | Artists#New       |  
+| /artists                              | POST       | Artists#Create    |  
+| /artists/:id/edit                     | GET        | Artists#Edit      |  
+| /artists/:id                          | PUT        | Artists#Update    |  
+| /artists/:id                          | DELETE     | Artists#Destroy   |  
+| /artists/:artistId/paintings          | GET        | Paintings#Index   |  
+| /artists/:artistId/paintings/:id      | GET        | Paintings#Show    |  
+| /artists/:artistId/paintings/new      | GET        | Paintings#New     |  
+| /artists/:artistsId/paintings         | POST       | Paintings#Create  |  
+| /artists/:artistId/paintings/:id/edit | GET        | Paintings#Edit    |  
+| /artists/:artistId/paintings/:id      | PUT        | Paintings#Update  |  
+| /artists/:artistId/paintings/:id      | DELETE     | Artists#Destroy   |  
+
+Note that the routes are _nested_ to reflect the database relationships that exists, but the controllers are not nested.
+
+</section>
+
+## Use TDD to Create a Movie's Review Page
+
+Today, we are going to use TDD to help us drive the development of our new feature!
+
+<section class='call-to-action' markdown='1'>
+
+With a partner, work on creating a test for the User Story below.  Make sure to consider:
+* What objects need to exist in the Arrange section?
+* How will you confirm what should be on the page?
+
+```
+As a user
+When I visit /movies/1/reviews
+Then I see
+    The title of the movie
+    All of that movie's reviews (with a rating and review content)
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### One Solution
+
+```c#
+[Fact]
```

Comment:
My read is that this exercise will be fairly difficult, I like giving them a change to try it. Might be nice to add an instructor note to fully work together as a class to build this test.